apps/site/assets/js/event-page-setup.js

```diff
@@ -29,9 +31,56 @@ function addInternetExplorerPolyfills() {
   stickyPolyfill();
 }
 
+function setupEventPopups() {
+  /**
+   * Why not use Bootstrap's popover? It's not equipped to handle having lots of
+   * interactive content, doesn't support a close button, and as implemented the
+   * UX for assistive tech users probably isn't ideal. For more, see "Making
+   * popovers work for keyboard and assistive technology users" in
+   * https://getbootstrap.com/docs/4.6/components/popovers/
+   *
+   * This implementation uses ally-dialog (https://a11y-dialog.netlify.app) as a
+   * base to create accessible dialog windows for various use cases.
+   */
+  const calendarView = document.querySelector(".m-events-hub--list-view");
+
+  // find all the event popup HTML elements
+  const eventPopups = calendarView.querySelectorAll(".m-event-overlay");
+  for (let i = 0; i < eventPopups.length; i++) {
+    const el = eventPopups[i];
+    const dialog = new A11yDialog(el);
+
+    // Hide popup when user clicks outside popup, or on a different event
+    document.addEventListener("click", e => {
```

Comment:
do these get removed at some point? You can listen for the turbolinks events to remove the listeners when the page is torn down.

Reply:
huh.  A lot of our JS scripts add event listeners, but I don't see them getting removed.  And I don't see a "turbolinks:destroy" or anything like that getting listened for anywhere else.  I assume we must be handling it differently, but I don't know how